### PR TITLE
[VBLOCKS-331] Fix `tokenWillExpire` event.

### PR DESCRIPTION
**Pull request by mhuynh5757**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [VBLOCKS-331](https://issues.corp.twilio.com/browse/VBLOCKS-331)

### Description

This PR fixes the event signature and cleans up unused event timeouts.

### Before review
* [x] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [x] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [x] Ready for review

### Before merge
* [x] Got one or more +1s
* [x] Squashed erroneous commits if necessary
* [x] Re-tested if necessary

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,8 +34,9 @@ const device = new Device(token, {
   tokenRefreshMs: 30000,
 });
 
-device.on('tokenWillExpire', () => {
-  return getTokenViaAjax().then(token => device.updateToken(token));
+device.on('tokenWillExpire', (dev: Device) => {
```
Comment (charliesantos) [inline CHANGELOG.md:37]: Why do we still need to pass the `device`'s reference to the handler if the user already have access to it?

Comment (mhuynh5757) [reply CHANGELOG.md:37]: It's not strictly necessary, but it matches the rest of the event signatures if we do pass the device to the callback.

Comment (ryan-rowland) [reply CHANGELOG.md:37]: I'd remove it from the example though, seems like we're promoting an antipattern

File: lib/twilio/device.ts [hunk 2]
```diff
@@ -1064,10 +1064,20 @@ class Device extends EventEmitter {
     this._publisher?.setHost(createEventGatewayURI(payload.home));
     if (payload.token) {
       this._identity = payload.token.identity;
-      this._tokenWillExpireTimeout = setTimeout(() => {
-        this.emit('tokenWillExpire');
-        this._tokenWillExpireTimeout = null;
-      }, payload.token.ttl - this._options.tokenRefreshMs!);
+      if (
+        typeof payload.token.ttl === 'number' &&
```
Comment (ryan-rowland) [inline lib/twilio/device.ts:1068]: Is this ever not a number?

File: lib/twilio/device.ts [hunk 3]
```diff
@@ -1064,10 +1064,20 @@ class Device extends EventEmitter {
...
+        typeof this._options.tokenRefreshMs === 'number'
```
Comment (ryan-rowland) [inline lib/twilio/device.ts:1069]: When is this not a number? What is the intention of this condition?

**Review by ryan-rowland — CHANGES_REQUESTED** [review]

**Pull request merged**